flink-queryable-state/flink-queryable-state-runtime/src/test/java/org/apache/flink/queryablestate/network/ClientTest.java

```diff
@@ -248,12 +251,30 @@ void testRequestUnavailableHost() throws Exception {
         try {
             client = new Client<>("Test Client", 1, serializer, stats);
 
-            InetSocketAddress serverAddress = new InetSocketAddress(InetAddress.getLocalHost(), 0);
-
-            KvStateInternalRequest request =
-                    new KvStateInternalRequest(new KvStateID(), new byte[0]);
-            CompletableFuture<KvStateResponse> future = client.sendRequest(serverAddress, request);
+            CompletableFuture<KvStateResponse> future;
+            Random random = new Random();
+            int numRetry = 0;
+            while (true) {
+                try {
+                    InetSocketAddress serverAddress =
+                            new InetSocketAddress(
+                                    InetAddress.getLocalHost(), random.nextInt(30000));
+
+                    KvStateInternalRequest request =
+                            new KvStateInternalRequest(new KvStateID(), new byte[0]);
+                    future = client.sendRequest(serverAddress, request);
+                    break;
+                } catch (Exception e) {
+                    if (ExceptionUtils.findThrowable(e, BindException.class).isPresent()
+                            && numRetry <= numTryCreateSocketAddress) {
+                        numRetry++;
+                    } else {
+                        throw e;
+                    }
+                }
+            }
```

Comment:
Actually, flink has NetUtils to find a free port 

`int port = NetUtils.getAvailablePort().getPort();`

Reply:
Good point. Fixed.